infra/main.tf

```diff
@@ -1,354 +1,147 @@
+  tags = local.tags
 }
 
-# Creating a load balancer
-resource "aws_alb" "front-lb" {
-  name               = "${var.app_name}-front-lb" # Naming our load balancer
-  load_balancer_type = "application"
-  subnets            = ["${aws_default_subnet.default_subnet_a.id}", "${aws_default_subnet.default_subnet_b.id}"]
 
-  # Referencing the security group
-  security_groups = ["${aws_security_group.aws-lb_security_group.id}"]
-}
+########################################################
+# EC2
+########################################################
 
-resource "aws_alb" "back-lb" {
-  name               = "${var.app_name}-back-lb" # Naming our load balancer
-  load_balancer_type = "application"
-  subnets            = ["${aws_default_subnet.default_subnet_a.id}", "${aws_default_subnet.default_subnet_b.id}"]
+module "ec2_instance" {
+  source  = "terraform-aws-modules/ec2-instance/aws"
+  version = "~> 5.0"
 
-  # Referencing the security group
-  security_groups = ["${aws_security_group.aws-lb_security_group.id}"]
-}
+  name          = "Stagnum-stack"
+  ami           = data.aws_ami.amazon_linux.id
+  instance_type = "t3.micro"
 
-# Creating a security group for the load balancer:
-resource "aws_security_group" "aws-lb_security_group" {
-  ingress {
-    from_port   = 80 # Allowing traffic in from port 80
-    to_port     = 80
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"] # Allowing traffic in from all sources
-  }
+  key_name                    = "deployer-key"
+  monitoring                  = true
+  vpc_security_group_ids      = [module.security_group.security_group_id]
+  subnet_id                   = element(module.vpc.public_subnets, 0)
+  availability_zone           = element(local.azs, 0)
+  associate_public_ip_address = true
+  user_data                   = templatefile("${path.root}/config/userdata.tftpl", local.data_inputs)
+  user_data_replace_on_change = true
 
-  egress {
-    from_port   = 0             # Allowing any incoming port
-    to_port     = 0             # Allowing any outgoing port
-    protocol    = "-1"          # Allowing any outgoing protocol 
-    cidr_blocks = ["0.0.0.0/0"] # Allowing traffic out to all IP addresses
-  }
+  tags       = local.tags
+  depends_on = [aws_key_pair.deployer]
 }
 
-# Creating a target group for the load balancer
-resource "aws_lb_target_group" "front-target-group" {
-  name        = "front-target-group"
-  port        = "80"
-  protocol    = "HTTP"
-  target_type = "ip"
-  vpc_id      = aws_default_vpc.default_vpc.id # Referencing the default VPC
-
-  health_check {
-    path                = "/"
-    interval            = 60
-    timeout             = 10
-    healthy_threshold   = 5
-    unhealthy_threshold = 5
-    matcher             = "200-399"
-  }
+resource "aws_key_pair" "deployer" {
+  key_name   = "deployer-key"
+  public_key = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOgP6TjSCjZS/VWhixYYevHGdzVN4jmlT5KH9va5CiBs elias.samuli@gmail.com"
```

Comment:
You might want to update this to be your public ssh key if you need access to the server